packages/dds/tree/src/test/simple-tree/api/eraseSchemaDetails.spec.ts

```diff
@@ -0,0 +1,409 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "node:assert";
+
+import {
+	TreeBeta,
+	TreeViewConfiguration,
+	allowUnused,
+	eraseSchemaDetails,
+	eraseSchemaDetailsSubclassable,
+	type InsertableTreeFieldFromImplicitField,
+	type InsertableTypedNode,
+	type InternalTreeNode,
+	type NodeFromSchema,
+	type TreeFieldFromImplicitField,
+	type TreeNode,
+	type TreeNodeSchema,
+	type WithType,
+	type typeSchemaSymbol,
+} from "../../../simple-tree/index.js";
+import {
+	SchemaFactory,
+	// eslint-disable-next-line import-x/no-internal-modules
+} from "../../../simple-tree/api/schemaFactory.js";
+
+import { getView } from "../../utils.js";
+import type { requireAssignableTo } from "@fluidframework/build-tools";
+import { Tree } from "../../../shared-tree/index.js";
+import type { areSafelyAssignable, requireTrue } from "../../../util/index.js";
+
+describe("eraseSchemaDetails", () => {
+	describe("eraseSchemaDetails", () => {
+		it("minimal", () => {
+			const schema = new SchemaFactory("com.example");
+
+			interface SquareNode {
+				readonly area: number;
+			}
+
+			interface SquareSchema {
+				create(sideLength: number): Square;
+			}
+
+			class SquareInternal
+				extends schema.object("Demo", { hidden: schema.number })
+				implements SquareNode
+			{
+				public get area(): number {
+					return this.hidden * this.hidden;
+				}
+
+				public static create(sideLength: number): SquareInternal {
+					return new SquareInternal({ hidden: sideLength });
+				}
+			}
+
+			const Square = eraseSchemaDetails<Square, SquareSchema>()(SquareInternal);
+			type Square = SquareNode & TreeNode & WithType<"com.example.Demo">;
+		});
+
+		it("recursive", () => {
+			const schema = new SchemaFactory("com.example");
+
+			interface SquareNode {
+				readonly area: number;
+			}
+
+			interface SquareSchema {
+				create(sideLength: number): Square;
+			}
+
+			class SquareInternal
+				extends schema.objectRecursive("Demo", {
+					hidden: [() => SquareInternal, SchemaFactory.number],
+				})
+				implements SquareNode
+			{
+				public get area(): number {
+					return this.hidden instanceof SquareInternal
+						? this.hidden.area
+						: this.hidden * this.hidden;
+				}
+
+				public static create(sideLength: number): SquareInternal {
+					return new SquareInternal({ hidden: sideLength });
+				}
+			}
+
+			const Square = eraseSchemaDetails<Square, SquareSchema>()(SquareInternal);
+			// This can't do `NodeFromSchema<typeof Square>` because TypeScript (but not intellisense) gives "Return type annotation circularly references itself".
```

Comment:
Nit: the "minimal" test case is missing this comment, which you seem to have included in the other cases here.